v4/internal/expect.go

```diff
@@ -91,21 +101,25 @@ type WantTxn struct {
 	NumErrors int
 }
 
+// WantSpan is a span or transaction expectation.
+type WantSpan struct {
+	Name       string
+	SpanID     string
```

Comment:
Just an idea:

Could we use `interface{}` as type for those attributes? `nil` could mean "any" value. And we could have separate types for constants (e. g. like `MatchNoParent = MatchNoParentType{}`).

Using `nil` for "any" value would help to avoid explicit initialization of those values in most cases.